app/src/main/java/org/fossasia/openevent/general/event/EventService.kt

```diff
@@ -21,6 +22,21 @@ class EventService(private val eventApi: EventApi, private val eventDao: EventDa
         }
     }
 
+    fun getSearchEvents(eventName: String): Flowable<List<Event>> {
+        val eventsFlowable = searchEventDao.getAllEvents()
+        return eventsFlowable.switchMap {
+            searchEventDao.deleteAll()
+            eventApi.searchEvents(eventName)
+                    .map {
+                        searchEventDao.insertEvents(it)
+                    }
+                    .toFlowable()
+                    .flatMap {
+                        eventsFlowable
+                    }
+        }
```

Comment:
The logic is wrong

Reply:
Could you please explain. I also thought something is wrong here.

Reply:
You are selecting and deleting all events from here on search. This should not be the case

Reply:
Should I create another function and delete the events there?